utils/src/main/kotlin/Utils.kt

```diff
@@ -392,6 +392,11 @@ fun Throwable.collectMessages(): List<String> {
     return messages
 }
 
+/**
+ * Recursively collect the messages of this [Throwable] and all its causes and join them to a single [String].
+ */
+fun Throwable.collectMessagesAsString() = collectMessages().joinToString("\nCaused by: ")
```

Comment:
Consider using a bit shorter name, e.g.: getMessagesString() or getMessagesAsString()

Reply:
I don't like using `get` when it is not a property accessor, that's why I chose `collect`.